src/lndk_offers.rs

```diff
@@ -98,6 +161,58 @@ pub(crate) async fn create_request_invoice(
     .unwrap()
 }
 
+// send_invoice_request sends the invoice request to node that created the offer we want to pay.
+pub async fn send_invoice_request(
+    network: Network,
+    invoice_request: InvoiceRequest,
+    onion_messenger: impl OnionMessageHandler + SendOnion,
+    intermediate_nodes: Vec<PublicKey>,
+    blinded_path: BlindedPath,
+    mut custom_msg_client: impl SendCustomMessage,
+) -> Result<(), OfferError<bitcoin::secp256k1::Error>> {
+    // We need to make sure our local onion messenger is connected to the needed local peers in
+    // so it can relay along the onion messages to them.
+    let onion_message_optional: u64 = 1 << ONION_MESSAGES_OPTIONAL;
+    let init_features = InitFeatures::from_le_bytes(onion_message_optional.to_le_bytes().to_vec());
+    let network = vec![ChainHash::using_genesis_block(network)];
+    onion_messenger
+        .peer_connected(
```

Comment:
Sounds good, so pretty much, connect to the introduction node, and have another `consume_messenger_events` loop running here that will be just be looking for the PeerConnected event and respond as needed?

Reply:
> have a consume_messenger_events loop running here

Not sure I understand why we'd have a second loop running here? Shouldn't we have the onion messenger running already if we're paying an offer to handling all of the sending/receiving?

Reply:
The onion_messenger instance we're using here in this cli/offers code is different from the one running when lndk is started up. So I don't think it'll receive those peer connect notifications? So another `consume_messenger_events` is what I thought you meant as far as not duplicating logic.
		
		However, I think the easiest thing would be to just, connect to the peer here as well as give the onion messenger this information. I don't think some duplicated code here would be too terrible, especially if we're going to get rid of it later anyway.
		
		And on that note, not sure if you saw my comment from above. We might not need the onion messenger instance here at all in the next release of LDK, IIUC https://github.com/lndk-org/lndk/pull/81#discussion_r1401560489

Reply:
> The onion_messenger instance we're using here in this cli/offers code is different from the one running when lndk is started up.

Right, of course. 

On a higher level, I think it would make sense to me that we pull out all of our onion messaging logic into one struct (ie, [run_onion_messenger](https://github.com/lndk-org/lndk/blob/master/src/onion_messenger.rs#L102) becomes its run function) like you've started to do with `SendOnion` and then surface `send_invoice_request` so that we don't have to duplicate some of the logic we already have (`send_onion` / `next_onion_message_for_peer` / `relay_outgoing_msg_event`). I hope that makes some sense, I'll take a proper look at the code and see if it works out the way I think it does.

Reply:
Opened this up to add some context: https://github.com/lndk-org/lndk/pull/85